src/markdown/generator.ml

```diff
@@ -0,0 +1,310 @@
+
+  let concat = List.fold_left ( ++ ) (Concat [])
+
+  let rec intersperse ~sep = function
+    | [] -> []
+    | [ h ] -> [ h ]
+    | h1 :: (_ :: _ as t) -> h1 :: sep :: intersperse ~sep t
+
+  let list ?(sep = Concat []) l = concat @@ intersperse ~sep l
+
+  let str fmt = Format.ksprintf (fun s -> String s) fmt
+
+  let escaped fmt = Format.ksprintf (fun s -> String s) fmt
+
+  let command s t = Command (s, t)
+
+  let rec pp fmt t =
+    match t with
+    | Inline s -> Format.fprintf fmt "%s" (String.concat "" s)
+    | Block b ->
+        let inner = function
+          | [] -> ()
+          | [ x ] -> Format.fprintf fmt "%a" pp x
+          | x :: xs -> Format.fprintf fmt "%a@\n%a" pp x pp (Block xs)
+        in
+        inner b
+    | Concat l -> pp_many fmt l
+    | Break -> Format.fprintf fmt "@\n"
+    | Anchor s -> Format.fprintf fmt "<a id=\"%s\"></a>" s
+    | String s -> Format.fprintf fmt "%s" s
+    | Env (indent, s, t, s') ->
+        Format.fprintf fmt "@[%s%s%a@]@,%s@]"
+          (match indent with Any -> "")
+          s pp t s'
+    | Command (s, t) -> Format.fprintf fmt "@[%s%a@]" s pp t
+    | OpenSqBracket -> Format.fprintf fmt "["
+    | CloseSqBracket -> Format.fprintf fmt "]"
+    | OpenParenthesis -> Format.fprintf fmt "("
+    | CloseParenthesis -> Format.fprintf fmt ")"
+
+  and pp_many fmt l = List.iter (pp fmt) l
+end
+
+open Markup
+
+let entity e =
+  match e with "#45" -> escaped "-" | "gt" -> str ">" | s -> str "&%s;" s
+
+let raw_markup (_ : Raw_markup.t) = noop
+
+let style (style : style) content =
+  match style with
+  | `Bold -> command "**" (content ++ str "**")
+  | `Italic | `Emphasis -> command "_" (content ++ str "_")
+  | `Superscript -> command "<sup>" content
+  | `Subscript -> command "<sub>" content
+
+let rec source_code (s : Source.t) =
+  match s with
+  | [] -> noop
+  | h :: t -> (
+      match h with
+      | Source.Elt i -> inline i ++ source_code t
+      | Tag (None, s) -> source_code s ++ source_code t
+      | Tag (Some _, s) -> source_code s ++ source_code t)
+
+and inline (l : Inline.t) =
+  match l with
+  | [] -> noop
+  | i :: rest -> (
+      match i.desc with
+      | Text "" -> inline rest
+      | Text _ ->
+          let l, _, rest =
+            Doctree.Take.until l ~classify:(function
+              | { Inline.desc = Text s; _ } -> Accum [ s ]
+              | _ -> Stop_and_keep)
+          in
+          str {|%s|} (String.concat "" l) ++ inline rest
+      | Entity e ->
+          let x = entity e in
+          x ++ inline rest
+      | Styled (sty, content) -> style sty (inline content) ++ inline rest
+      | Linebreak -> break ++ inline rest
+      | Link (href, content) ->
+          (let rec f (content : Inline.t) =
+             match content with
+             | [] -> noop
+             | i :: rest -> (
+                 match i.desc with
+                 | Text s ->
+                     command "" (str "[%s]" s ++ str "(%s)" href) ++ f rest
```

Comment:
```suggestion
                     command "" (str "[%s](%s)" s href) ++ f rest
```

Reply:
This is resolved in the most recent commit, thanks!